proxy.go

```diff
@@ -102,6 +102,21 @@ func NewProxy(config Config) *Proxy {
 	return proxy
 }
 
+// SetTimeout set Timeout of the proxy.
+func (p *Proxy) SetTimeout(timeout time.Duration) {
+	p.timeout = timeout
+}
+
+// SetTransport set RoundTripper of the proxy.
+func (p *Proxy) SetTransport(transport *http.Transport) {
+	p.transport = transport
+}
+
+// SetDial set Dialer of the proxy.
+func (p *Proxy) SetDial(dial *net.Dialer) {
```

Comment:
It'd be better to use an interface here and not just `net.Dialer`, makes it easier to supply custom implementation.

The comment should explain when this dialer is actually used.

Reply:
There is no interface to net.Dialer in the standard lib to my knowledge, we can create one in this repo but I don't really see the point.